pkg/operator/management/management_state_controller.go

```diff
@@ -0,0 +1,157 @@
+package management
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+
+	configv1helpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	"github.com/openshift/library-go/pkg/operator/events"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+var workQueueKey = "instance"
+
+type ManagementStateController struct {
+	operatorName          string
+	operatorClient        operatorv1helpers.OperatorClient
+	clusterOperatorGetter configv1client.ClusterOperatorsGetter
+	eventRecorder         events.Recorder
+
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+}
+
+func NewOperatorManagementStateController(
+	name string,
+	clusterOperatorClient configv1client.ClusterOperatorsGetter,
+	operatorStatusProvider operatorv1helpers.OperatorClient,
+	recorder events.Recorder,
+) *ManagementStateController {
+	c := &ManagementStateController{
+		operatorName:          name,
+		clusterOperatorGetter: clusterOperatorClient,
+		operatorClient:        operatorStatusProvider,
+		eventRecorder:         recorder,
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ManagementStateController-"+name),
+	}
+
+	operatorStatusProvider.Informer().AddEventHandler(c.eventHandler())
+	// TODO watch clusterOperator.status changes when it moves to openshift/api
+
+	return c
+}
+
+// sync reacts to a change in prereqs by finding information that is required to match another value in the cluster. This
```

Comment:
?

Reply:
copy & pasta from status_controller, I will refit this